src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2040 @@
+
+        # TODO check if already running....otherwise start multiple apps
+        # TODO  -> in acquisition mode it looks like it does not start a second app, but also does not report that
+        # TODO -> in processing mode it is possible to start multiple apps....
+        # TODO find out where to ask for acq or processing mode
+        # TODO is there a clever way for checking if app still running? Seems to be no command available to check
+        # TODO appEnd only works for the last opened window
+        # TODO want to check if we want to start app invisible (sVisible = False)
+
+        # self.AppStart() # start HPDTA software  # TODO for testing in order to not start a new App
+
+        if children:
+            try:
+                kwargs = children["readoutcam"]
+            except Exception:
+                raise
+            self._readoutcam = ReadoutCamera(parent=self, spectrograph=children.get("spectrograph"),
+                                             daemon=daemon, **kwargs)
+            self.children.value.add(self._readoutcam)  # add readoutcam to children-VA
+            try:
+                kwargs = children["streakunit"]
+            except Exception:
+                raise
+            self._streakunit = StreakUnit(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._streakunit)  # add streakunit to children-VA
+            try:
+                kwargs = children["delaybox"]
+            except Exception:
+                raise
+            self._delaybox = DelayGenerator(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._delaybox)  # add delaybox to children-VA
+
+    def _openConnection(self):
+        """
+        open connection with RemoteEx client.
+        :parameter host: IP-adress or hostname
+        :parameter port: port for sending/receiving commands
+        :parameter port_d: port for reading images
+        return: connection to RemoteEx command and data port
+        """
+        # connect to sockets
+        try:
+            self._commandport = socket.create_connection((self.host, self.port), timeout=5)
+            self._dataport = socket.create_connection((self.host, self.port_d), timeout=5)
+        except socket.timeout as msg:
+            raise model.HwError(msg, "Failed to connect to '%s using port %d'. Check the server "
+                                "is connected to the network, turned "
+                                " on, and correctly configured." % (self.host, self.port))
+        except socket.error as msg:
+            raise model.HwError(msg, "Failed to connect to '%s:%d'. Check ...." % (self.host, self.port))
+
+        # check if connection returns correct response
+        try:
+            message = self._commandport.recv(self.port)
+            if message != 'RemoteEx Ready\r':
+                raise ValueError("Connection to port %s not successfull. "
```

Comment:
If the other end answers unexpectedly, raise `IOError`, instead of ValueError, which is more typically if the function was called with the wrong parameters.